js.go

```diff
@@ -2751,10 +2753,18 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 					// wait this time.
 					noWait = false
 					err = sendReq()
-				} else if err == ErrTimeout && len(msgs) == 0 {
+				} else if err == ErrTimeout {
 					// If we get a 408, we will bail if we already collected some
-					// messages, otherwise ignore and go back calling NextMsg.
-					err = nil
+					// messages, otherwise ignore and go back calling nextMsg.
+					if len(msgs) == 0 {
+						err = nil
+						continue
+					}
+					subjectParts := strings.Split(msg.Subject, ".")
+					// ignore timeout message from server if it comes from a different pull request
+					if reqID != "" && subjectParts[len(subjectParts)-1] != reqID {
```

Comment:
I know that subject should never be empty, but in case of some bug elsewhere, would it be good to check for len, maybe with some small helper?

```
func subjectMatchesReqID(subject, reqID string) bool {
    subjectParts := strings.Split(subject, ".")
    if len(subjectParts) < 2 {
        return false
    }
    return subjectParts[len(subjectParts)-1] == reqID
}
```

Reply:
Even for empty string, `strings.Split()` will return a slice with a single element, so we'll not panic here. I can add that check anyway though.

Reply:
A helper function will be good regardless.